### PR TITLE
DATAUP-519: Add cell IDs to the job comm channel init code

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Add the IDs of the current narrative cells to the generated python code that starts `jobcomm.py` in the narrative backend.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-519
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and seeing that nothing has changed. How boring!

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/common/jobCommChannel.js [hunk 1]
```diff
@@ -576,9 +576,16 @@ define([
...
             return [
                 'from biokbase.narrative.jobs.jobcomm import JobComm',
-                'JobComm().start_job_status_loop(init_jobs=True)',
+                'cell_list = ' +
+                    JSON.stringify(
+                        currentCells.map((cell) => {
+                            return cell.cell_id;
+                        })
+                    ),
+                'JobComm().start_job_status_loop(cell_list=cell_list, init_jobs=True)',
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobCommChannel.js:588]: start the job status loop with a list of cell IDs in the current narrative

File: test/unit/spec/kbaseNarrativeSpec.js [hunk 2]
```diff
@@ -65,6 +65,7 @@ define([
             Jupyter.notebook = {
                 _fully_loaded: DEFAULT_FULLY_LOADED,
                 writable: DEFAULT_WRITABLE,
+                get_cells: () => [],
```
Comment (ialarmedalien) [inline test/unit/spec/kbaseNarrativeSpec.js:68]: needed when kicking off the narrative

File: kbase-extension/static/kbase/js/common/jobCommChannel.js (hunk 1 above)
Comment (briehl) [reply kbase-extension/static/kbase/js/common/jobCommChannel.js:588]: Those are the Jupyter-made cell ids, not the ones that live in KBase metadata. Those would be under `cell.metadata.kbase.attributes.id`. Jupyter cell ids aren't always the same on restart. Unless that's what you mean?

Comment (briehl) [reply kbase-extension/static/kbase/js/common/jobCommChannel.js:588]: The KBase cell ids are what are sent along with job metadata.

**Review by briehl — CHANGES_REQUESTED** [review]
Just a question about how cell ids should be used.

File: kbase-extension/static/kbase/js/common/jobCommChannel.js (hunk 1 above)
Comment (ialarmedalien) [reply kbase-extension/static/kbase/js/common/jobCommChannel.js:588]: Doh... why have one unique cell ID when you could have several?! OK, will fix.

File: test/unit/spec/kbaseNarrativeSpec.js [hunk 3]
```diff
@@ -62,41 +61,18 @@ define([
...
+
+            Jupyter.notebook = Mocks.buildMockNotebook({
+                fullyLoaded: DEFAULT_FULLY_LOADED,
+                readOnly: !DEFAULT_WRITABLE,
+                commInfoReturn: {
+                    comms: {
+                        some_comm_id: {
+                            target_name: 'KBaseJobs',
+                        },
                     },
```
Comment (ialarmedalien) [inline test/unit/spec/kbaseNarrativeSpec.js:73]: transfer all this stuff to the mock and use that instead.

**Comment by ialarmedalien** [comment]
Looks like something is going wrong in the UI module with modals. 😞

**Review by briehl — APPROVED** [review]
LGTM

**Pull request merged**